File: ProjC1/src/projc1/Game/Position.java
package projc1.Game;

class Position {
    private Point coordinate; // COMPOSICIÓN: Position contiene Point
    
    public Position(int x, int y) {
        this.coordinate = new Point(x, y); // Point se crea con Position
    }
    
    public int getX() { return coordinate.getX(); }
    public int getY() { return coordinate.getY(); }
    
    public void move(int deltaX, int deltaY) {
        coordinate.setX(coordinate.getX() + deltaX);
        coordinate.setY(coordinate.getY() + deltaY);
    }
    
    public void setPosition(int x, int y) {
        coordinate.setX(x);
        coordinate.setY(y);
    }
    
    public double distanceTo(Position other) {
        return this.coordinate.distanceTo(other.coordinate);
    }
}

File: ProcC1_pyver/game/components/position.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    
    def get_x(self): return self.x
    def get_y(self): return self.y
    def set_x(self, x): self.x = x
    def set_y(self, y): self.y = y
    
    def distance_to(self, other):
        return ((other.x - self.x)**2 + (other.y - self.y)**2)**0.5


class Position:
    def __init__(self, x, y):
        self.coordinate = Point(x, y)  # Composición
    
    def get_x(self): return self.coordinate.get_x()
    def get_y(self): return self.coordinate.get_y()
    
    def move(self, delta_x, delta_y):
        self.coordinate.set_x(self.coordinate.get_x() + delta_x)
        self.coordinate.set_y(self.coordinate.get_y() + delta_y)
    
    def set_position(self, x, y):
        self.coordinate.set_x(x)
        self.coordinate.set_y(y)
    
    def distance_to(self, other):
        return self.coordinate.distance_to(other.coordinate)